MSOE.MediaComplete.Lib/Library/DataSource/FileSystem.cs

```diff
@@ -567,17 +574,27 @@ private static bool IsFileLocked(FileInfo file)
         /// <param name="song"></param>
         private void UpdateFile(LocalSong song)
         {
-            var tagFile = TaglibFile.Create(song.Path);
-
-            var tag = tagFile.Tag;
-            _cachedSongs[song.Id].Title = tag.Title;
-            _cachedSongs[song.Id].Artists = tag.AlbumArtists;
-            _cachedSongs[song.Id].Album = tag.Album;
-            _cachedSongs[song.Id].Genres = tag.Genres;
-            _cachedSongs[song.Id].Year = tag.Year;
-            _cachedSongs[song.Id].TrackNumber = tag.Track;
-            _cachedSongs[song.Id].SupportingArtists = tag.Performers;
-            // Duration is assumed to be fixed
+            try
+            {
+                var tagFile = TaglibFile.Create(song.Path);
+                var tag = tagFile.Tag;
+                _cachedSongs[song.Id].Title = tag.Title;
+                _cachedSongs[song.Id].Artists = tag.AlbumArtists;
+                _cachedSongs[song.Id].Album = tag.Album;
+                _cachedSongs[song.Id].Genres = tag.Genres;
+                _cachedSongs[song.Id].Year = tag.Year;
+                _cachedSongs[song.Id].TrackNumber = tag.Track;
+                _cachedSongs[song.Id].SupportingArtists = tag.Performers;
+                // Duration is assumed to be fixed
+            }
+            catch (FileNotFoundException)
+            {
+
```

Comment:
as a result of how the changed-renamed-create-delete filewatcher events happen, sometimes we get commands to update a file that has already moved, and therefore it causes an exception. as such, it is unhandled.